src/test/rpc/Book_test.cpp

```diff
@@ -1072,10 +1072,33 @@ class Book_test : public beast::unit_test::suite
         BEAST_EXPECT(jv[jss::status] == "success");
     }
 
+    // Check that a stream only sees the given OfferCreate once
+    static
+    bool
+    offerOnlyOnceInStream(
+        std::unique_ptr<WSClient>& wsc,
+        std::chrono::milliseconds const& timeout,
+        jtx::PrettyAmount takerGets,
+        jtx::PrettyAmount takerPays)
+    {
+        auto maybeJv = wsc->getMsg(timeout);
+        // No message
+        if (!maybeJv)
+            return false;
+        // wrong message
+        auto const& t = (*maybeJv)[jss::transaction];
```

Comment:
if `jss::transaction` is not a member, the subsequent lines will assert, so you might just want to explicitly return false for that case.